collect_app/src/main/java/org/odk/collect/android/activities/FillBlankFormActivity.kt

```diff
@@ -0,0 +1,351 @@
+                }
+
+                override fun denied() {}
+            }
+        )
+    }
+
+    override fun onResume() {
+        super.onResume()
+        if (formSyncTask != null) {
+            formSyncTask!!.setDiskSyncListener(this)
+            if (formSyncTask!!.status == AsyncTask.Status.FINISHED) {
+                syncComplete(formSyncTask!!.statusMessage)
+            }
+        }
+    }
+
+    override fun onPause() {
+        if (formSyncTask != null) {
+            formSyncTask!!.setDiskSyncListener(null)
+        }
+        super.onPause()
+    }
+
+    /**
+     * Called by DiskSyncTask when the task is finished
+     */
+    override fun syncComplete(result: String) {
+        Timber.i("Disk scan complete")
+        hideProgressBarAndAllow()
+        showSnackbar(result)
+    }
+
+    private fun setupAdapter() {
+        val columnNames = arrayOf(
+            DatabaseFormColumns.DISPLAY_NAME,
+            DatabaseFormColumns.JR_VERSION,
+            DatabaseFormColumns.DATE,
+            DatabaseFormColumns.GEOMETRY_XPATH
+        )
+        val viewIds = intArrayOf(
+            R.id.form_title,
+            R.id.form_subtitle,
+            R.id.form_subtitle2,
+            R.id.map_view
+        )
+        listAdapter = FormListAdapter(
+            listView,
+            DatabaseFormColumns.JR_VERSION,
+            this,
+            R.layout.form_chooser_list_item,
+            { parent: AdapterView<*>?, view: View?, position: Int, id: Long ->
+                onMapButtonClick(
+                    id
+                )
+            },
+            columnNames,
+            viewIds
+        )
+        listView.adapter = listAdapter
+    }
+
+    override fun getSortingOrderKey(): String {
+        return FORM_CHOOSER_LIST_SORTING_ORDER
+    }
+
+    override fun updateAdapter() {
+        supportLoaderManager.restartLoader(LOADER_ID, null, this)
+    }
+
+    override fun onCreateLoader(id: Int, args: Bundle?): Loader<Cursor> {
+        showProgressBar()
+        return CursorLoaderFactory(currentProjectProvider).getFormsCursorLoader(
+            filterText,
+            sortingOrder,
+            hideOldFormVersions()
+        )
+    }
+
+    override fun onLoadFinished(loader: Loader<Cursor>, cursor: Cursor) {
+        hideProgressBarIfAllowed()
+        listAdapter.swapCursor(cursor)
+    }
+
+    override fun onLoaderReset(loader: Loader<Cursor>) {
+        listAdapter.swapCursor(null)
+    }
+
+    private fun hideOldFormVersions(): Boolean {
+        return settingsProvider.getUnprotectedSettings()
+            .getBoolean(ProjectKeys.KEY_HIDE_OLD_FORM_VERSIONS)
+    }
+
+    companion object {
+        private const val FORM_CHOOSER_LIST_SORTING_ORDER = "formChooserListSortingOrder"
+    }
+}
+
+class SelectFormFromMap : SelectItemFromMap<Input>() {
+
+    override fun createIntent(context: Context, input: Input): Intent {
+        return Intent(context, FormMapActivity::class.java).also {
```

Comment:
If you use apply instead of also you can call parameters/functions directly like (without `it.` or `intent.`):

```
putExtra(FormMapActivity.EXTRA_FORM_ID, formId)
            putExtra(
                SelectionMapActivity.EXTRA_NEW_ITEM,
                Intent(context, FormEntryActivity::class.java).apply {
                    action = Intent.ACTION_EDIT
                    data = FormsContract.getUri(projectId, formId)
                }
            )
```

Reply:
I tend to prefer `also` over `apply` as it I'm not really a fan of the magic used to make it work (receiver added to a lambda signature). I feel like every time I see `apply` I have to go work out how it works under the hood 😆.

Reply:
I think it's ok not to understand how some methods that we use work under the hood. We believe they work well if they are provided for use. For the same reason we don't write test for functions from libraries - we believe they are tested.

Reply:
> For the same reason we don't write test for functions from libraries - we believe they are tested.

Agreed. If I can reword what I mean, I find the lack of `it` (or an explicit parameter) distracting for two reasons:

1. It seems unclear what the receiver is
2. If I (correctly) assume the receiver is the subject of `apply` call then it's unclear how it works to me (I basically forget that you can add a receiver type to function literals)

I definitely don't think it is bad form to use `apply`, but I personally prefer `also` or `let` as I find those more readable. Does that make sense?

Reply:
Ok if you prefer `also` that's fine you will probably find more `apply` in my pull request but that is not a problem.